### ISSUE TITLE
[enhancement]: List `eject` as a suggested/recommended dependency for the deb

### ISSUE THREAD

**Issue by Monochromics** [canonical/cloud-init#4732]
# Enhancement/Bug

The `eject` package is used in the azure datasource: https://github.com/canonical/cloud-init/blob/cb36bf38b823f811a3e938ccffc03d7d13190095/cloudinit/sources/helpers/azure.py#L790

`eject` is not listed as a dependency or a recommendation/suggestion of the deb.  It look's like this is the only datasource that uses this, so listing as a full dep might be a bit heavy handed. This isn't typically and issue, as most images already include eject, however ubuntu minimal images susceptible. 

I'd also argue that the current handling of the missing package is ungraceful (even if mostly benign).

```
2024-01-03 19:58:43,765 - subp.py[DEBUG]: Running command ['eject', '/dev/sr0'] with allowed return codes [0] (shell=False, capture=True)
2024-01-03 19:58:43,767 - azure.py[DEBUG]: Failed ejecting the provisioning iso: Unexpected error while running command.
```

**Comment by j5awry** [comment]
I think we need to gather info about the availability of the command for all supported distros

1. is the command guaranteed to be installed?
2. if so, how is it guaranteed (for instance, in Ubuntu, it'd be in a seed file _or_ as a package dependency)
3. it not, how do we solve this cross-distro?

For more background, we're hitting this in Ubuntu because, previously, `eject` was being supplied as a `recommends` of another package. in Ubuntu Minimal, we no longer install any `recommended` packages. We hit this with other `cloud-init` dependencies while working on the new Minimal seeds, but this is the first time we're seeing a requirement that's in a single cloud. for dependencies used across clouds, Ubuntu has added packages to our Minimal seed as required. But we're hesitant to include a package for a _single_ cloud in a definition of a base image. We're not against adding it directly to our Azure image build _but_, before we take that route, we want to ensure we're not the only ones hitting the issue. 

tl:dr -- if it's just Ubuntu, we can add it directly to our image builds. If it may affect other distros as well, I think we should find a different solution than all of us adding it to our Azure builds.

**Comment by catmsred** [comment]
Since there is no way to restrict dependencies (or recommends/suggests) to a single cloud, would it make sense to include the dependency as a commented out suggestion in the debian packaging?  That would make it clear to anyone experiencing difficulty getting cloud-init to run on Azure what they need to do to fix the issue but also not add bloat to cloud-init installs in non-Azure environment.

**Comment by j5awry** [comment]
another alternative suggested is `/lib/udev/cdrom_id`

this will be available on all `systemd` based distributions

**Comment by igalic** [comment]
(In FreeBSD we would use `camcontrol eject`, which should be in the `PATH`)
